abracadabra/src/g4-mandatory/trajectory.hh

```diff
@@ -19,48 +19,46 @@ public:
-    void SetEnergyDeposit(G4double);
-    void SetFinalVolume  (G4String);
-    void SetFinalProcess (G4String);
-
-    virtual G4VTrajectoryPoint* GetPoint(G4int i) const;
-    virtual int  GetPointEntries() const;
-    virtual void AppendStep(const G4Step*);
-    virtual void MergeTrajectory(G4VTrajectory*);
-    virtual void ShowTrajectory(std::ostream&) const;
-    virtual void DrawTrajectory() const;
+    G4String              GetParticleName()    const override { return pdef_->GetParticleName(); }
+    G4int                 GetPDGEncoding()     const override { return pdef_->GetPDGEncoding(); }
+    G4double              GetCharge()          const override { return pdef_->GetPDGCharge(); }
+    G4VTrajectoryPoint*   GetPoint(G4int i)    const override { return (*trjpoints_)[i]; }
+    int                   GetPointEntries()    const override { return trjpoints_->size(); }
+    G4ThreeVector         GetInitialMomentum() const override { return initial_momentum_; }
+    G4int                 GetTrackID()         const override { return trackId_; }
+    G4int                 GetParentID()        const override { return parentId_; }
+
+    void ShowTrajectory (std::ostream& os) const override { G4VTrajectory::ShowTrajectory(os); }
+    void DrawTrajectory ()                 const override { G4VTrajectory::DrawTrajectory(); }
+    void AppendStep     (const G4Step*)          override;
+    void MergeTrajectory(G4VTrajectory*)         override;
+
+    G4ParticleDefinition* GetParticleDefinition() const { return pdef_; }
+    G4double              GetInitialTime()        const { return initial_time_; }
+    G4String              GetInitialVolume()      const { return initial_volume_; }
+    G4ThreeVector         GetInitialPosition()    const { return initial_position_; }
+    G4ThreeVector         GetFinalMomentum()      const { return final_momentum_; }
+    G4ThreeVector         GetFinalPosition()      const { return final_position_; }
+    G4double              GetFinalTime()          const { return final_time_; }
+    G4String              GetFinalProcess()       const { return final_process_; }
+    G4String              GetFinalVolume()        const { return final_volume_; }
+    G4double              GetTrackLength()        const { return length_; }
+    G4double              GetEnergyDeposit()      const { return edep_; }
+    G4String              GetCreatorProcess()     const { return creator_process_; }
+
+
+    void SetFinalMomentum(const G4ThreeVector& m) { final_momentum_ = m; }
+    void SetFinalPosition(const G4ThreeVector& x) { final_position_ = x; }
+    void SetFinalTime    (G4double t)             { final_time_ = t; }
+    void SetTrackLength  (G4double l)             { length_ = l; }
+    void SetEnergyDeposit(G4double e)             { edep_ = e; }
+    void SetFinalProcess (G4String fp)            { final_process_ = fp; }
+    void SetFinalVolume  (G4String fv)            { final_volume_ = fv; }
```

Comment:
No `override` implies that these are not G4-imposed, so names should be snake_case.

On might argue that, for the sake of consistency, they should be CamelCase ... but I suspect that highlighting the distinction between standard G4 and our additions might be worth more. Unless you feel strongly about it, please change them to snake_case. (But, actually, as I'll discuss elsewhere, I think they shouldn't be here at all, until after we have completed the first milesone.)